src/Aspire.Hosting.JavaScript/JavaScriptHostingExtensions.cs

```diff
@@ -370,7 +370,14 @@ public static IResourceBuilder<ViteAppResource> AddViteApp(this IDistributedAppl
             runScriptName,
             argsCallback: c =>
             {
-                c.Args.Add("--");
+                // pnpm does not strip the -- separator and passes it to the script, causing Vite to ignore subsequent arguments.
+                // npm and yarn both strip the -- separator before passing arguments to the script.
+                // Only add the separator for npm and yarn.
+                if (c.Resource.TryGetLastAnnotation<JavaScriptPackageManagerAnnotation>(out var packageManager) &&
+                    packageManager.ExecutableName != "pnpm")
+                {
+                    c.Args.Add("--");
+                }
```

Comment:
Let's not do it this way. Instead the JavaScriptPackageManagerAnnotation should have something on it to say whether it should do this or not.

I initially had a `string? CommandSeparator { get; } = "--";` but removed it because I figured all package managers would be OK with `--`. But if pnpm doens't work with this, then I guess we will need it.

Reply:
No worries! (Had just woken up so was about to take a look at this but happy you've done it instead 😄 )